### ISSUE TITLE
NSwagStudio not autodetecting .NET 5 open API doc via DI properly

### ISSUE THREAD

**Issue by bzuillsmith** [RicoSuter/NSwag#3322]
In .NET 5.0, it seems there is a new method for registering and serving the open API document through swagger. Now loading this project from NSwagStudio does not work. It does not pick up the injected swagger document service via auto-detection and the manual method seems deprecated. I'm not sure how to use it.

The error we get is `No service for type 'NSwag.Generation.IOpenApiDocumentGenerator' has been registered.` for 

    services.AddSwaggerGen(c =>
    {
        c.SwaggerDoc("v1", new OpenApiInfo { Title = "MyAPI", Version = "v1" });
    });

**Comment by RicoSuter** [comment]
This is the Swashbuckle registration (now sadly for NSwag the default in .NET 5). This of course does not work with nswagstudio. You need to replace swashbuckle with nswag.aspnetcore if you want to use this in flow in the ui.

**Comment by bzuillsmith** [comment]
Ahhh, I see. I did not realize they were separate projects. Thought there was some direct relationship or partnership. We figured out enough to just use the msbuild generation method and that seems to work well. We were just trying to experiment in nswagstudio to understand all the csharp generator settings. We'll compare features, ease of use, and decide if we want to fully switch to nswag or swagger, but for now it's working with a bit of both.
